### PR TITLE
More compile fixes when not having uint128_t

### PR DESCRIPTION
**Pull request by herwinw**
The typecast fixes did make me long for a language with operator overloading.

### CODE REVIEW

File: src/lib/util/uint128.h [hunk 1]
```diff
@@ -62,7 +62,7 @@
 /** Create a 128 bit integer value with n bits high
  *
  */
-static uint128_t uint128_gen_mask(uint8_t bits)
+UNUSED static uint128_t uint128_gen_mask(uint8_t bits)
```
Comment (alandekok) [inline src/lib/util/uint128.h]: it's better to just `ifdef` them out... I thought it already did that?

Comment (arr2036) [reply src/lib/util/uint128.h]: the compiler definitely will not warn about unused functions defined in headers.

Comment (arr2036) [reply src/lib/util/uint128.h]: maybe change them to static inline if you're seeing issues

Comment (herwinw) [reply src/lib/util/uint128.h]: `inline` did the trick, I thought that was a C++ thing that didn't exist in C

**Comment by arr2036** [comment]
I don't think you can just do the assignment.  There needs to be something to explicitly copy the low 64bit to a 64bit integer.

**Comment by arr2036** [comment]
Closed by b271e352f5edffbdfe5ff18bc17be9bb15371cf1

**Pull request closed without merging**